### DISCUSSION TITLE
STM32F303VCTx/Discovery, can't get a serial monitor output via usb st-link

### DISCUSSION THREAD

**Discussion by warriormaster12** [stm32duino/Arduino_Core_STM32#1902] [Q&A] [answered]
This may be due to my lack of understanding my board but I still want to know how to fix this issue. Basically, I can upload my program to the board and it works how it should so no problems there. After that I went to _DigitalReadSerial_ example and I tried to get the program to print at least a "hello world" and nothing. As far as I understand, I should be able to get an output when using st-link usb but maybe I'm wrong. How should I fix this issue? I've tried looking online but results that I have found haven't helped me what so ever.

```
OS: Nobara 36/Fedora 36

Application: Arduino IDE V2, Flatpak

Port: /dev/ttyS0

Upload Method: STM32CubeProgrammer(SWD)

USB: Generic Serial enabled
```

**Comment by fpistm** [comment, marked as answer]
Hi @warriormaster12 

Which  version of the PCB you get?
BAsed on user manual:
![image](https://user-images.githubusercontent.com/20641798/208907771-565c7379-c154-4b6b-ab62-a0c3fefcf9ea.png)

So if you have a version A or B you have to add SB.

**Reply by warriormaster12** [reply]
How do I check which version I have and how to add SB in my code?

**Reply by fpistm** [reply]
You should have a sticker on bottom side of your Board with something like `MB1035 x` where x is the version (letter).
About SB, this is not linked to the code but to hardware. It means Solder Bridge as explain in the User Manual:
![image](https://user-images.githubusercontent.com/20641798/208913412-dd16a253-bcd5-4922-86ca-095d25f71e56.png)

On bottom side of your board you should see SB13 and SB15, if there is no 0 Ohms resistor then it explains why you didn't get anything on the VCP.

**Reply by warriormaster12** [reply]
Okay, I have MB1035 B and there doesn't seem to be connected anything to it. So what would be the workaround for this use a regular St-link?

![image](https://user-images.githubusercontent.com/33091666/208915468-d20ed3b7-fa9e-4973-8af0-a423b75af387.png)

**Reply by fpistm** [reply]
Well you have to shorten each SB, you have to do a tin solder. Ideally with a 0 ohms resistor or a small wire.

**Reply by warriormaster12** [reply]
Damn, so this applies both to the usb port and st-link port?

**Reply by fpistm** [reply]
Don't get your point.
Please, Read the User Manual of the board which explain that. To sum up the USART (RX/TX) of the MCU is not connected (wired) to the STLink then you don't have data on VCP created by the STLink. To connect them simply feed with tin solder the each  SB.

**Reply by warriormaster12** [reply]
Aight, thanks 👍
